server/src/main/java/org/candlepin/dto/api/v1/KeycloakStatusDTO.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2009 - 2018 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api.v1;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+/**
+ * Keycloak parameters needed in status endpoint when enabled in configuration file
+ */
+public class KeycloakStatusDTO extends StatusDTO {
+
+    private String keycloakRealm;
+    private String keycloakAuthUrl;
+    private String keycloakResource;
+
+
+    public String getKeycloakResource() {
+        return keycloakResource;
+    }
+
+    public KeycloakStatusDTO setKeycloakResource(String keycloakResource) {
+        this.keycloakResource = keycloakResource;
+        return this;
+    }
+
+    public String getKeycloakAuthUrl() {
+        return keycloakAuthUrl;
+    }
+
+    public StatusDTO setKeycloakAuthUrl(String keycloakAuthUrl) {
+        this.keycloakAuthUrl = keycloakAuthUrl;
+        return this;
+    }
+
+    public String getKeycloakRealm() {
+        return keycloakRealm;
+    }
+
+    public StatusDTO setKeycloakRealm(String keycloakRealm) {
+        this.keycloakRealm = keycloakRealm;
+        return this;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Most of this should be replaced by a call to the base equals check via super.equals(obj). Take a look at any of the existing DTOs that extend TimestampedCandlepinDTO for an example.